---
title: "Drought shifts sorghum root metabolite and microbiome profiles and enriches the stress response factor pipecolic acid"
author: "Daniel Caddell"
date: "08/03/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown  
This document includes analyses of the 16S amplicon sequencing and metabolomic profiling of root, rhizosphere, and soil from the EPICON/CSP-summer2017 field season.

## load packages
```{r}
library(dada2);packageVersion("dada2")
library(phyloseq); packageVersion("phyloseq")
library(ape);packageVersion("ape")
library(data.table);packageVersion("data.table")
library(ggplot2); packageVersion("ggplot2")
library(reshape2); packageVersion("reshape2")
library(dplyr); packageVersion("dplyr")
library(RColorBrewer); packageVersion("RColorBrewer")
library(beepr); packageVersion("beepr")
library(scales); packageVersion("scales")
library(grid); packageVersion("grid")
library(ggalt); packageVersion("ggalt")
library(mctoolsr); packageVersion("mctoolsr")
library(vegan); packageVersion("vegan")
library(agricolae);packageVersion("agricolae")
library(patchwork);packageVersion("patchwork")
library(ggfortify);packageVersion("ggfortify")
theme_set(theme_bw()) #Define a default theme for ggplot graphics.

#setwd("/set/working/directory/location/")
setwd("/users/dcaddell/Desktop/")
```

## import OTU biom file, sample file, and tree file, then build phyloseq object  
update file locations with correct paths
```{r}
biom_file <- "~/Desktop/qiime2_output/merged-feature-table.biom" #update path info
map_file <- "~/Desktop/qiime2_output/sample-metadata-final_feature.tsv" #update path info
tree_file <- "~/Desktop/qiime2_output/tree.nwk" #update path info
otutax <- import_biom(biom_file, parseFunction = parse_taxonomy_default)
sam <- import_qiime_sample_data(map_file)
tree <- read.tree(tree_file)
all <- merge_phyloseq(otutax, sam, tree)
all
colnames(tax_table(all)) <- c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")
colnames(tax_table(all))
```

## Remove mitochondrial and chloroplast reads from a phyloseq object (Illumina - greengenes).  
A full list of related functions are in 'functions_for16SData.Rmd'.  
WARNING: only use on taxonomies generated by greengenes, otherwise pattern matching will not find mito/chloro ASVs.
This chunk has been modified from code originally written by Alex Styer.
```{r}
remove_mitoAndChloro <- function(x){#where x is a merged phyloseq object
  df <- as.data.frame(tax_table(x))
  setDT(df, keep.rownames = TRUE)
  colnames(df)[1] <- "ASV"
  allASVs <- df[,1]
  mito <- filter(df, df$Family == "f__mitochondria")
  chloro <- filter(df, df$Class == "c__Chloroplast")
  contam <- as.vector(rbind(mito, chloro)$ASV)
  ASVs_toKeep <- as.vector(subset(allASVs, !(ASV %in% contam))$ASV)
  num_mito <- nrow(mito)
  num_chloro <- nrow(chloro)
  cat(sprintf("%i and %s ASVs were removed as pesky mitochondria or chloroplast, respectively.\nKhaleesi frees these bacterial slaves to the wild binary yonder.", num_mito, num_chloro))
  x <- prune_taxa(ASVs_toKeep, x)
}
```

## Clean up taxonomy
Takes a psmelted phyloseq object and cleans up greengenes-format taxonomies.  
i.e. removes the "k__" in k__Bacteria.  
Also takes NA values and unassigned taxa and renames their taxonomy as "Unknown".
This chunk has been modified from code originally written by Alex Styer.
```{r}
cleanup_taxonomy <- function(x){ #to be used on dfs made from melted phyloseq objects
  x$Domain <- as.character(lapply(x$Domain, sub, pattern = "k__", replacement = ""))
  x$Phylum <- as.character(lapply(x$Phylum, sub, pattern = "p__", replacement = ""))
  x$Class <- as.character(lapply(x$Class, sub, pattern = "c__", replacement = ""))
  x$Order <- as.character(lapply(x$Order, sub, pattern = "o__", replacement = ""))
  x$Family <- as.character(lapply(x$Family, sub, pattern = "f__", replacement = ""))
  x$Genus <- as.character(lapply(x$Genus, sub, pattern = "g__", replacement = ""))
  x$Species <- as.character(lapply(x$Species, sub, pattern = "s__", replacement = ""))
  x$Domain <- ifelse(x$Domain == "", "Unknown", as.character(x$Domain))
  x$Phylum <- ifelse(x$Phylum == "", "Unknown", as.character(x$Phylum))
  x$Class <- ifelse(x$Class == "", "Unknown", as.character(x$Class))
  x$Order <- ifelse(x$Order == "", "Unknown", as.character(x$Order))
  x$Family <- ifelse(x$Family == "", "Unknown", as.character(x$Family))
  x$Genus <- ifelse(x$Genus == "", "Unknown", as.character(x$Genus))
  x$Species <- ifelse(x$Species == "", "Unknown", as.character(x$Species))
  x$Domain <- ifelse(is.na(x$Domain) == TRUE , "Unknown", as.character(x$Domain))
  x$Phylum <- ifelse(is.na(x$Phylum) == TRUE , "Unknown", as.character(x$Phylum))
  x$Class <- ifelse(is.na(x$Class) == TRUE , "Unknown", as.character(x$Class))
  x$Order <- ifelse(is.na(x$Order) == TRUE , "Unknown", as.character(x$Order))
  x$Family <- ifelse(is.na(x$Family) == TRUE , "Unknown", as.character(x$Family))
  x$Genus <- ifelse(is.na(x$Genus) == TRUE , "Unknown", as.character(x$Genus))
  x$Species <- ifelse(is.na(x$Species) == TRUE , "Unknown", as.character(x$Species))
  x <- x
}
```

## Additional Phyloseq tools  

A quantitative/informative tool for deciding which number of top ASVs to keep for stacked bar plots.  
Takes a phyloseq object and plots each ASV's relative abundance against its rank from 1-N (1 being most abundant, N least).  
Also prints to console summary statistics decribing the number of ASVs that represent a given fraction of readcounts.
This chunk has been modified from code originally written by Alex Styer.
```{r}
plot_relAbund_curve <- function(x){
  df <- as.data.frame(rowSums(otu_table(x)))
  setDT(df, keep.rownames = TRUE)
  colnames(df) <- c("ASV", "ASV_sum")
  all_sum <- sum(df$ASV_sum)
  df$ASV_relAbund <- df$ASV_sum/all_sum
  df <- df[order(-ASV_relAbund)]
  df$ASV_rank <- 1:nrow(df)
  
  #a function to get percentiles
  get_percentile <- function(y, n){#n is a double between 0-1; the output will tell you the smallest number of ASVs that account for that perecnt relative abundance in your dataset.
    row = 1
    num = 0
    while(num<=n){
      num = sum(y$ASV_relAbund[1:row])
      row = row + 1
    }
    y$ASV_rank[row-1]
  }
  
  #a function to print percentiles to console
  print_percentiles <- function(a,b,c,d,e,f,g,h,i){
    cat(sprintf("the top %i ASVs = ~10%% of all reads
the top %i ASVs = ~20%% of all reads
the top %i ASVs = ~30%% of all reads
the top %i ASVs = ~40%% of all reads
the top %i ASVs = ~50%% of all reads
the top %i ASVs = ~60%% of all reads
the top %i ASVs = ~70%% of all reads
the top %i ASVs = ~80%% of all reads
the top %i ASVs = ~90%% of all reads"
                ,a,b,c,d,e,f,g,h,i))
  }
  
  #get summary statistics
  mean_relabund <- mean(df$ASV_relAbund)
  per10 <- get_percentile(df, .1)
  per20 <- get_percentile(df, .2)
  per25 <- get_percentile(df, .25)
  per30 <- get_percentile(df, .3)
  per40 <- get_percentile(df, .4)
  per50 <- get_percentile(df, .5)
  per60 <- get_percentile(df, .6)
  per70 <- get_percentile(df, .7)
  per75 <- get_percentile(df, .75)
  per80 <- get_percentile(df, .8)
  per90 <- get_percentile(df, .9)
  
  #set plot aesthetics and add-ons
  x_title <- scale_x_continuous(name = "ASV Rank", breaks = seq(0, max(df$ASV_rank), by = 10))
  y_title <- scale_y_continuous(name = "ASV Relative Abundance")
  vline25 <- geom_vline(xintercept = per25, color="#444546")
  vline25_text <- geom_text(aes(x=per25, y=(max(df$ASV_relAbund)/2), label=(sprintf("top %i ASVs = ~25%% reads", per25))), vjust=-1, angle=90, size=10/.pt, color="#444546", check_overlap = TRUE)
  vline50 <- geom_vline(xintercept = per50, color="#444546")
  vline50_text <- geom_text(aes(x=per50, y=(max(df$ASV_relAbund)/2), label=(sprintf("top %i ASVs = ~50%% reads", per50))), vjust=-1, angle=90, size=10/.pt, color="#444546", check_overlap = TRUE)
  vline75 <- geom_vline(xintercept = per75, color="#444546")
  vline75_text <- geom_text(aes(x=per75, y=(max(df$ASV_relAbund)/2), label=(sprintf("top %i ASVs = ~75%% reads", per75))), vjust=-1, angle=90, size=10/.pt,color="#444546", check_overlap = TRUE)
  hline_mean <- geom_hline(yintercept = mean_relabund, color="#a04344")
  hline_mean_text <- geom_text(aes(x=(max(df$ASV_rank)/2), y=mean_relabund, label=(sprintf("mean relative abundance = %f", mean_relabund))), vjust = -1, size=10/.pt,color="#a04344", check_overlap = TRUE)

  #plot ASV rank by relative abundance
  if (nrow(df)<=300) {
    print_percentiles(per10, per20, per30, per40, per50, per60, per70, per80, per90)
    ggplot(df, aes(x=ASV_rank, y=ASV_relAbund)) + 
      geom_point() +
      list(x_title, y_title, vline25, vline25_text, vline50, vline50_text, vline75, vline75_text, hline_mean, hline_mean_text)
  }else{
    if (nrow(df)>300 && per75<=300) {
      df <- df[1:300]
      cat("Trimming data to top 300 ASVs for plot readability\nSummary statistics represent all input ASVs\n")
      print_percentiles(per10, per20, per30, per40, per50, per60, per70, per80, per90)
      ggplot(df, aes(x=ASV_rank, y=ASV_relAbund)) + 
        geom_point() +
        list(x_title, y_title, vline25, vline25_text, vline50, vline50_text, vline75, vline75_text, hline_mean, hline_mean_text)
    }else{
      if (nrow(df)>300 && per50<=300 && per75>300) {
        df <- df[1:300]
        cat("Trimming data to top 300 ASVs for plot readability\nSummary statistics represent all input ASVs\n")
        print_percentiles(per10, per20, per30, per40, per50, per60, per70, per80, per90)
        ggplot(df, aes(x=ASV_rank, y=ASV_relAbund)) + 
          geom_point() +
          list(x_title, y_title, vline25, vline25_text, vline50, vline50_text, hline_mean, hline_mean_text)
      }else{
        if (nrow(df)>300 && per25<=300 && per50>300) {
          df <- df[1:300]
          cat("Trimming data to top 300 ASVs for plot readability\nSummary statistics represent all input ASVs\n")
          print_percentiles(per10, per20, per30, per40, per50, per60, per70, per80, per90)
          ggplot(df, aes(x=ASV_rank, y=ASV_relAbund)) + 
            geom_point() +
            list(x_title, y_title, vline25, vline25_text, hline_mean, hline_mean_text)
        }else{
          df <- df[1:300]
          cat("Trimming data to top 300 ASVs for plot readability\nSummary statistics represent all input ASVs\n")
          print_percentiles(per10, per20, per30, per40, per50, per60, per70, per80, per90)
          ggplot(df, aes(x=ASV_rank, y=ASV_relAbund)) + 
            geom_point() +
            list(x_title, y_title, hline_mean, hline_mean_text)
        }
      }
    }
  } 
}
```

Now run the cleanup functions to remove any remaining mitochondria and chloroplast sequences.  
Also melt phyloseq object into a dataframe and transform sample counts for further analyses.  
```{r}
all.clean <- remove_mitoAndChloro(all)
all.clean.df <- psmelt (all.clean)
all.clean.df <- cleanup_taxonomy(all.clean.df)
#transform data
all.trans <- transform_sample_counts(all.clean, function(OTU) 100*OTU/sum(OTU))
#merge groups within transformed data
all.trans.merge <- merge_samples(all.trans, "Group")
all.trans.merge <- transform_sample_counts(all.trans.merge, function(OTU) 100*OTU/sum(OTU))
all.tm.df<- psmelt(all.trans.merge)
all.tm.df <- cleanup_taxonomy(all.tm.df) #NOTE: some columns were messed up during merge_samples
beep(4)
```

## Save the session so that it can be loaded later without having to rerun this step.
This will save time compared to reruning the code above.
```{r eval=FALSE, include=TRUE}
save.image("~/Desktop/Illumina/phyloseq_illuminaASV.rdata")
#close R, re-open R
load("~/Desktop/Illumina/phyloseq_illuminaASV.rdata")
```

Check the total reads per sample, and the distribution of reads.
```{r}
readsumsdf = data.frame(nreads = sort(taxa_sums(all.clean), TRUE), sorted = 1:ntaxa(all.clean), 
                        type = "OTUs")
readsumsdf = rbind(readsumsdf, data.frame(nreads = sort(sample_sums(all.clean), 
                                                        TRUE), sorted = 1:nsamples(all), type = "Samples"))
title = "Total number of reads"
p = ggplot(readsumsdf, aes(x = sorted, y = nreads)) + geom_bar(stat = "identity")
p + ggtitle(title) + scale_y_log10() + facet_wrap(~type, 1, scales = "free")

plot_relAbund_curve(all.clean) # this function is from 'functions_for16SData.Rmd'
```

## Alpha Diversity   
Note: this analysis is performed using rarefied data, which differs from the transformation method performed on the data above.
```{r}
set.seed(42)
all_rare<-rarefy_even_depth(all.clean, rngseed = TRUE)
#sanity check
par(mfrow = c(1, 2))
title = "Sum of reads for each sample, all_rare"
plot(sort(sample_sums(all_rare), TRUE), type = "h", main = title, ylab = "reads") 
#, ylim = c(0, 1000))

richness = estimate_richness(all_rare,measures = "Shannon")
richness
```

ANOVA of alpha-diversity and Tukey-HSD posthoc test
```{r}
temp.aov <- sample_data(all_rare)
temp.aov <- temp.aov[,c(4:9)]
temp.aov$ST <- paste0(temp.aov$SampleType, temp.aov$Timepoint) #add column to see if sample diversity changes with timepoint
data.aov <- cbind(temp.aov, richness)
data.aov
#ANOVA posthoc test: Tukey-HSD
all_aov <- aov(Shannon ~ Group, data = data.aov)
summary(all_aov)
all_aov_tukey1 <- TukeyHSD(all_aov, 'Group', conf.level=0.95)
all_aov_tukey1 #Significance between watered and drought roots

#ANOVA posthoc test: Tukey-HSD
all_aov <- aov(Shannon ~ SampleType, data = data.aov)
summary(all_aov)
all_aov_tukey2 <- TukeyHSD(all_aov, 'SampleType', conf.level=0.95)
all_aov_tukey2 #Significance between roots and rhizosphere/soil
```

Plot of Shannon Diversity Index, Figure 1C
```{r}
p1c <- plot_richness(all_rare, x="SampleType", measures=c("Shannon"), color="Group", shape="Treatment", title="Shannon") +
        geom_point( size = 7) +
        scale_x_discrete(limits = c("Root","Rhizosphere","Soil"), labels = c("Root","Rhizosphere","Soil")) +
        scale_color_manual(values=c("#4d5b6a","#686a47","#cb9e59",
                                    "#4d5b6a","#686a47","#cb9e59", 
                                    "#7997a1","#919f70", "#ecc363",
                                    "#7997a1","#919f70", "#ecc363"))+
        theme(axis.text.x=element_text(size=11,color="black",angle=0), 
              axis.text.y=element_text(size=11,color="black"), 
              axis.title=element_text(size=11,face="bold"), 
              text=element_text(size=11, face="bold"),
              plot.title=element_text(size=15,hjust= 0.5, vjust=2),
              legend.title=element_text(size=10), 
              legend.text=element_text(size=8))
p1c
#ggsave("figure_1c.pdf", plot = p1c, width=5, height=5.5, useDingbats=FALSE)
#ggsave("Shannon.png", plot = p1c, width=5, height=5.5, dpi=600)
```

## Beta Diversity
Beta diversity (PCoA) of sorghum root, rhizosphere, and soil microbiomes at TP8 and 24 hours after rewatering in well-watered control or drought plots. Figure 1D
```{r}
ord.mds.bray <- ordinate(all.trans, method="MDS", distance="bray")
#Plot MDS
p1d <- phyloseq::plot_ordination(all.trans, ord.mds.bray, color="Group", shape="Treatment", title="Bray-Curtis distance")+
          geom_vline(xintercept = 0,linetype="longdash",color="gray")+
          geom_hline(yintercept = 0,linetype="longdash",color="gray")+
          geom_point( size = 7) +
          scale_color_manual(values=c("#4d5b6a","#686a47","#cb9e59",
                                      "#4d5b6a","#686a47","#cb9e59", 
                                      "#7997a1","#919f70", "#ecc363",
                                      "#7997a1","#919f70", "#ecc363"))+
          theme(axis.text.x=element_text(size=11,color="black",angle=0), 
                axis.text.y=element_text(size=11,color="black"), 
                axis.title=element_text(size=11,face="bold"), 
                text=element_text(size=11, face="bold"),
                plot.title=element_text(size=15,hjust= 0.5, vjust=2),
                legend.title=element_text(size=10), 
                legend.text=element_text(size=8))
p1d
#ggsave("figure_1d.pdf", plot = p1d, width=7, height=5.5, useDingbats=FALSE)
#ggsave("figure_1d.png", plot = p1d, width=7, height=5.5, dpi=600)
```

Adonis test and pairwise permanova posthoc test
```{r}
ps.prop_dist <- distance(all.trans, method = "bray")
ps.prop_data <- data.frame(sample_data(all.trans))
summary(ps.prop_data)
#test <- adonis(ps.prop_dist ~ SampleType + Treatment + Rep + Date + Timepoint, data = ps.prop_data)
test <- adonis(ps.prop_dist ~ SampleType + Treatment + Rep + Date + Group, data = ps.prop_data)
test
# get test results as a table
test_tab <- test$aov.tab
test_tab_rows <- rownames(test_tab)
test_tab_rows
for(i in 1:6){ #change "1:n" to number of rows being assessed
  a1 <- c("The percent of variance attributable to the factor")
  b2 <- toString(test_tab_rows[i])
  c3 <- c("is")
  d4 <- toString(round(test_tab$SumsOfSqs[i]/test_tab$SumsOfSqs[7], 3)) #change [n] to the row of "Total"
  print(paste(a1,b2,c3,d4))
}
#adonis posthoc test: Pairwise permanova
calc_pairwise_permanovas(ps.prop_dist, ps.prop_data, "Treatment")
calc_pairwise_permanovas(ps.prop_dist, ps.prop_data, "SampleType")
calc_pairwise_permanovas(ps.prop_dist, ps.prop_data, "SampleTreatment")
```

## Relative abundance
I renamed the samples, and set the order that they will appear (x-axis) for figures.  
```{r}
#Separate replicates
SampleOrder <-   scale_x_discrete(limits = c("DC13","DC15","DC18",#TP8, control, root
                                             "DC29","DC31","DC34",#TP8_24h, control, root                              
                                             "DC14","DC16","DC17", #TP8, drought, root
                                             "DC30","DC32","DC33", #TP8_24h, drought, root
                                            "DC2","DC3","DC6",#TP8, control, rhizosphere
                                             "DC19","DC21","DC24", #TP8_24h, control, rhizosphere                     
                                             "DC1","DC4","DC5",#TP8, drought, rhizosphere
                                             "DC20","DC22","DC23", #TP8_24h, drought, rhizosphere    
                                              "DC7","DC9","DC12",#TP8, control, soil
                                             "DC25","DC27","DC36", #TP8_24h, control, soil                            
                                             "DC8","DC10","DC11",#TP8, drought, soil
                                             "DC26","DC28","DC35"), #TP8_24h, drought, soil    
                                  labels = c( "TP8_Control_Root","TP8_Control_Root","TP8_Control_Root",
                                             "24h_Control_Root","24h_Control_Root","24h_Control_Root",                
                                             "TP8_Drought_Root","TP8_Drought_Root","TP8_Drought_Root",
                                             "24h_Watered_Root","24h_Watered_Root","24h_Watered_Root",
                                        "TP8_Control_Rhizosphere","TP8_Control_Rhizosphere","TP8_Control_Rhizosphere",
                                        "24h_Control_Rhizosphere","24h_Control_Rhizosphere","24h_Control_Rhizosphere",
                                        "TP8_Drought_Rhizosphere","TP8_Drought_Rhizosphere","TP8_Drought_Rhizosphere",
                                        "24h_Watered_Rhizosphere","24h_Watered_Rhizosphere","24h_Watered_Rhizosphere",
                                             "TP8_Control_Soil","TP8_Control_Soil","TP8_Control_Soil",
                                             "24h_Control_Soil","24h_Control_Soil","24h_Control_Soil",                
                                             "TP8_Drought_Soil","TP8_Drought_Soil","TP8_Drought_Soil",
                                             "24h_Watered_Soil","24h_Watered_Soil","24h_Watered_Soil"
                                              ))
#merged replicates
ReplicateMerge <- scale_x_discrete(limits = c( "TP8_Watered_Root",
                                           "TP24h_Watered_Root",                           
                                           "TP8_Drought_Root",
                                           "TP24h_Drought_Root",
                                           "TP8_Watered_Rhizosphere",
                                           "TP24h_Watered_Rhizosphere",                           
                                           "TP8_Drought_Rhizosphere",
                                           "TP24h_Drought_Rhizosphere",      
                                          "TP8_Watered_Soil",
                                           "TP24h_Watered_Soil",                            
                                           "TP8_Drought_Soil",
                                           "TP24h_Drought_Soil"),
                                labels = c( "TP8_Control_Root",
                                           "24h_Control_Root",                             
                                           "TP8_Drought_Root",
                                           "24h_Watered_Root",
                                            "TP8_Control_Rhizosphere",
                                           "24h_Control_Rhizosphere",                             
                                           "TP8_Drought_Rhizosphere",
                                           "24h_Watered_Rhizosphere",                                 
                                           "TP8_Control_Soil",
                                           "24h_Control_Soil",                            
                                           "TP8_Drought_Soil",
                                           "24h_Watered_Soil"
                                          ))
```

Custom color palette created by Alex Styer, based on Farrow and Ball paints, used for distinguishing bacterial phyla.
```{r}
farrowAndBall_palette <- c(
   "#4d5b6a" #Stiffkey Blue
  ,"#6a90b4" #Cook's Blue
  ,"#599ec4" #ST Giles Blue
  ,"#a1c5c8" #Blue Ground
  ,"#7997a1" #Stone Blue
  ,"#427e83" #Vardo
  ,"#84b59c" #Arsenic
  ,"#919f70" #Yeabridge Green
  ,"#686a47" #Bancha
  ,"#c8bd83" #Churlish Green
  ,"#cb9e59" #India Yellow
  ,"#ecc363" #Babouche
  ,"#c57b67" #Red Earth
  ,"#d65f3d" #Charlotte's Locks
  ,"#a04344" #Incarnadine
  ,"#bf7a8f" #Rangwali
  ,"#8d8089" #Brassica
  ,"#50414c" #Pelt
  ,"#e5e0db" #Strong White
  ,"#444546" #Off-Black
)
farrowAndBall_pal <- function() scales::manual_pal(farrowAndBall_palette)
scale_fill_farrowAndBall <- function(...) ggplot2::discrete_scale('fill', 'farrowAndBall', farrowAndBall_pal(), ...)
```

Phylum level relative abundances of sorghum root, rhizosphere, and soil microbiomes at TP8 and 24 hours after rewatering (24h DW) in well-watered (W) or drought (D) plots. Figure 1B
```{r message=FALSE, warning=FALSE}
phylum_list <- dplyr::select(all.tm.df, Abundance,Phylum)
phylum_list <- phylum_list %>% group_by(Phylum) %>% summarise(Abundance = sum(Abundance))
phylum_list <- phylum_list[order(-phylum_list$Abundance),]
phylum_T10 <- dplyr::slice(phylum_list, 1:10)

'%ni%' <- Negate('%in%')
all.tm.df$Phylum <- as.character(all.tm.df$Phylum)
all.tm.df[all.tm.df$Phylum %ni% phylum_T10$Phylum,]$Phylum <-'Other'
all.tm.df$Phylum <- as.factor(all.tm.df$Phylum)
all.tm.df$Phylum <- factor(all.tm.df$Phylum, levels = rev(c(phylum_T10$Phylum,"Other")))

farrowAndBall_ManPal <- c(
  "#4d5b6a" #Stiffkey Blue
  ,"#599ec4" #ST Giles Blue
  ,"#a1c5c8" #Blue Ground
  ,"#7997a1" #Stone Blue
  ,"#427e83" #Vardo
  ,"#84b59c" #Arsenic
  ,"#919f70" #Yeabridge Green
  ,"#686a47" #Bancha
  ,"#cb9e59" #India Yellow
  ,"#a04344" #Incarnadine
  ,"#50414c") #Pelt

p1b <- ggplot(all.tm.df, aes(x=Sample, y=Abundance, fill=Phylum))+
        geom_bar(stat = "identity", position ="stack", width = 0.95)+
        ReplicateMerge +
        scale_fill_manual(values = farrowAndBall_ManPal) +
        ggtitle("Illumina MiSeq ASVs") +     
        theme(axis.text.x=element_text(angle = 270, hjust=0, vjust=0.5),
              plot.title = element_text(lineheight=.8, face="bold"))
p1b
#ggsave("figure_1b.pdf", plot = p1b, width=5, height=5.5, useDingbats=FALSE)
#ggsave("figure_1b.png", plot = p1b, width=5, height=5.5, dpi=600)
```

Relative abundances of individual lineages that displayed a significant difference in abundance between watering treatments (ANOVA, Tukey-HSD, P<0.05), as in Figures 1E-1J. For reference, phyla that did not show a significant abundance pattern are also included.

Actinobacteria RA, Figure 1E
```{r}
#Subset Actinobacteria
all.Actino <- subset_taxa(all.trans, Phylum %in% c("p__Actinobacteria"))
all.Actino.ASV.temp <- psmelt(all.Actino)
all.Actino.ASV <- subset(all.Actino.ASV.temp, Abundance > 0)
length(unique(all.Actino.ASV$OTU))

Actino.temp <- as.data.frame(paste(all.Actino.ASV$Group,all.Actino.ASV$Rep,sep="_"))
Actino.temp$Abundance <- all.Actino.ASV$Abundance
Actino.temp$OTU <- all.Actino.ASV$OTU
colnames(Actino.temp)[1] <- "Group"
Actino.RA <- Actino.temp%>%
  group_by(Group)%>%
  summarise(total=sum(Abundance))
Actino.RA
#Remove replicate names from Group
Actino.RA <- data.frame(lapply(Actino.RA, function(x) { gsub("_1", "", x) }))
Actino.RA <- data.frame(lapply(Actino.RA, function(x) { gsub("_2", "", x) }))
Actino.RA <- data.frame(lapply(Actino.RA, function(x) { gsub("_3", "", x) }))
Actino.RA[,2] <- as.numeric(as.character(Actino.RA[,2]))
Actino.RA
#write.csv(Actino.RA,"Actino.summary.csv")

#Perform ANOVA and TukeyHSD posthoc
Actino_aov <- aov(total ~ Group, data = Actino.RA)
summary(Actino_aov)
Actino_tukey <- HSD.test(Actino_aov, 'Group')
Actino_tukey

#Plot Actinobacteria relative abundance
p.Actino<- ggplot(data=Actino.RA, aes(x=Group, y=total, fill=Group))+
  ReplicateMerge +
  geom_boxplot(fill="#50414c")+
  geom_dotplot(binaxis='y',fill="black", stackdir='center', stackratio=1, dotsize=1.25)+
  ylim(0,100)+
  labs(title = "Actinobacteria", x = "Group", y = "Abundance") +
  theme_bw(base_size = 20) +
  theme(axis.text.x=element_text(size=12,color="black",angle=90), 
        axis.text.y=element_text(size=12,color="black"), 
        axis.title=element_text(size=12,face="bold"),
        text=element_text(size=12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5))
p.Actino

#ggsave("Actino_sub.pdf", plot = p.Actino, width=5, height=6, useDingbats=FALSE)
#ggsave("Actino_sub.png", plot = p.Actino, width=5, height=6, dpi=600)
```

Proteobacteria RA, Figure 1G
```{r}
#Subset Proteobacteria
all.Proteo <- subset_taxa(all.trans, Phylum %in% c("p__Proteobacteria"))
all.Proteo.ASV.temp <- psmelt(all.Proteo)
all.Proteo.ASV <- subset(all.Proteo.ASV.temp, Abundance > 0)
length(unique(all.Proteo.ASV$OTU))

Proteo.temp <- as.data.frame(paste(all.Proteo.ASV$Group,all.Proteo.ASV$Rep,sep="_"))
Proteo.temp$Abundance <- all.Proteo.ASV$Abundance
Proteo.temp$OTU <- all.Proteo.ASV$OTU
colnames(Proteo.temp)[1] <- "Group"
Proteo.RA <- Proteo.temp%>%
  group_by(Group)%>%
  summarise(total=sum(Abundance))
Proteo.RA
#Remove replicate names from Group
Proteo.RA <- data.frame(lapply(Proteo.RA, function(x) { gsub("_1", "", x) }))
Proteo.RA <- data.frame(lapply(Proteo.RA, function(x) { gsub("_2", "", x) }))
Proteo.RA <- data.frame(lapply(Proteo.RA, function(x) { gsub("_3", "", x) }))
Proteo.RA[,2] <- as.numeric(as.character(Proteo.RA[,2]))
Proteo.RA
#write.csv(Proteo.RA,"Proteo.summary.csv")

#Perform ANOVA and TukeyHSD posthoc
Proteo_aov <- aov(total ~ Group, data = Proteo.RA)
summary(Proteo_aov)
Proteo_tukey <- HSD.test(Proteo_aov, 'Group')
Proteo_tukey

#Plot Proteobacteria relative abundance
p.Proteo<- ggplot(data=Proteo.RA, aes(x=Group, y=total, fill=Group))+
  ReplicateMerge +
  geom_boxplot(fill="#a04344")+
  geom_dotplot(binaxis='y',fill="black", stackdir='center', stackratio=1, dotsize=1.25)+
  ylim(0,70)+
  labs(title = "Proteobacteria", x = "Group", y = "Abundance") +
  theme_bw(base_size = 20) +
  theme(axis.text.x=element_text(size=12,color="black",angle=90), 
        axis.text.y=element_text(size=12,color="black"), 
        axis.title=element_text(size=12,face="bold"),
        text=element_text(size=12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5))
p.Proteo

#ggsave("Proteo_sub.pdf", plot = p.Proteo, width=5, height=6, useDingbats=FALSE)
#ggsave("Proteo_sub.png", plot = p.Proteo, width=5, height=6, dpi=600)
```

Acidobacteria RA
```{r}
#Subset Acidobacteria
all.Acido <- subset_taxa(all.trans, Phylum %in% c("p__Acidobacteria"))
all.Acido.ASV.temp <- psmelt(all.Acido)
all.Acido.ASV <- subset(all.Acido.ASV.temp, Abundance > 0)
length(unique(all.Acido.ASV$OTU))

Acido.temp <- as.data.frame(paste(all.Acido.ASV$Group,all.Acido.ASV$Rep,sep="_"))
Acido.temp$Abundance <- all.Acido.ASV$Abundance
Acido.temp$OTU <- all.Acido.ASV$OTU
colnames(Acido.temp)[1] <- "Group"
Acido.RA <- Acido.temp%>%
  group_by(Group)%>%
  summarise(total=sum(Abundance))
Acido.RA
#Remove replicate names from Group
Acido.RA <- data.frame(lapply(Acido.RA, function(x) { gsub("_1", "", x) }))
Acido.RA <- data.frame(lapply(Acido.RA, function(x) { gsub("_2", "", x) }))
Acido.RA <- data.frame(lapply(Acido.RA, function(x) { gsub("_3", "", x) }))
Acido.RA[,2] <- as.numeric(as.character(Acido.RA[,2]))
Acido.RA
#write.csv(Acido.RA,"Acido.summary.csv")

#Perform ANOVA and TukeyHSD posthoc
Acido_aov <- aov(total ~ Group, data = Acido.RA)
summary(Acido_aov)
Acido_tukey <- HSD.test(Acido_aov, 'Group')
Acido_tukey

#Plot Acidobacteria relative abundance
p.Acido<- ggplot(data=Acido.RA, aes(x=Group, y=total, fill=Group))+
  ReplicateMerge +
  geom_boxplot(fill="#cb9e59")+
  geom_dotplot(binaxis='y',fill="black", stackdir='center', stackratio=1, dotsize=1.25)+
  ylim(0,35)+
  labs(title = "Acidobacteria", x = "Group", y = "Abundance") +
  theme_bw(base_size = 20) +
  theme(axis.text.x=element_text(size=12,color="black",angle=90), 
        axis.text.y=element_text(size=12,color="black"), 
        axis.title=element_text(size=12,face="bold"),
        text=element_text(size=12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5))
p.Acido
```

Chloroflexi RA
```{r}
#Subset Chloroflexi
all.Chloroflexi <- subset_taxa(all.trans, Phylum %in% c("p__Chloroflexi"))
all.Chloroflexi.ASV.temp <- psmelt(all.Chloroflexi)
all.Chloroflexi.ASV <- subset(all.Chloroflexi.ASV.temp, Abundance > 0)
length(unique(all.Chloroflexi.ASV$OTU))

Chloroflexi.temp <- as.data.frame(paste(all.Chloroflexi.ASV$Group,all.Chloroflexi.ASV$Rep,sep="_"))
Chloroflexi.temp$Abundance <- all.Chloroflexi.ASV$Abundance
Chloroflexi.temp$OTU <- all.Chloroflexi.ASV$OTU
colnames(Chloroflexi.temp)[1] <- "Group"
Chloroflexi.RA <- Chloroflexi.temp%>%
  group_by(Group)%>%
  summarise(total=sum(Abundance))
Chloroflexi.RA
#Remove replicate names from Group
Chloroflexi.RA <- data.frame(lapply(Chloroflexi.RA, function(x) { gsub("_1", "", x) }))
Chloroflexi.RA <- data.frame(lapply(Chloroflexi.RA, function(x) { gsub("_2", "", x) }))
Chloroflexi.RA <- data.frame(lapply(Chloroflexi.RA, function(x) { gsub("_3", "", x) }))
Chloroflexi.RA[,2] <- as.numeric(as.character(Chloroflexi.RA[,2]))
Chloroflexi.RA
#write.csv(Chloroflexi.RA,"Chloroflexi.summary.csv")

#Perform ANOVA and TukeyHSD posthoc
Chloroflexi_aov <- aov(total ~ Group, data = Chloroflexi.RA)
summary(Chloroflexi_aov)
Chloroflexi_tukey <- HSD.test(Chloroflexi_aov, 'Group')
Chloroflexi_tukey

#Plot Chloroflexi relative abundance
p.Chloroflexi<- ggplot(data=Chloroflexi.RA, aes(x=Group, y=total, fill=Group))+
  ReplicateMerge +
  geom_boxplot(fill="#686a47")+
  geom_dotplot(binaxis='y',fill="black", stackdir='center', stackratio=1, dotsize=1.25)+
  ylim(0,20)+
  labs(title = "Chloroflexi", x = "Group", y = "Abundance") +
  theme_bw(base_size = 20) +
  theme(axis.text.x=element_text(size=12,color="black",angle=90), 
        axis.text.y=element_text(size=12,color="black"), 
        axis.title=element_text(size=12,face="bold"),
        text=element_text(size=12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5))
p.Chloroflexi
```

Bacteroidetes RA, Figure 1H
```{r}
#Subset Bacteroidetes
all.Bacteroidetes <- subset_taxa(all.trans, Phylum %in% c("p__Bacteroidetes"))
all.Bacteroidetes.ASV.temp <- psmelt(all.Bacteroidetes)
all.Bacteroidetes.ASV <- subset(all.Bacteroidetes.ASV.temp, Abundance > 0)
length(unique(all.Bacteroidetes.ASV$OTU))

Bacteroidetes.temp <- as.data.frame(paste(all.Bacteroidetes.ASV$Group,all.Bacteroidetes.ASV$Rep,sep="_"))
Bacteroidetes.temp$Abundance <- all.Bacteroidetes.ASV$Abundance
Bacteroidetes.temp$OTU <- all.Bacteroidetes.ASV$OTU
colnames(Bacteroidetes.temp)[1] <- "Group"
Bacteroidetes.RA <- Bacteroidetes.temp%>%
  group_by(Group)%>%
  summarise(total=sum(Abundance))
Bacteroidetes.RA
#Remove replicate names from Group
Bacteroidetes.RA <- data.frame(lapply(Bacteroidetes.RA, function(x) { gsub("_1", "", x) }))
Bacteroidetes.RA <- data.frame(lapply(Bacteroidetes.RA, function(x) { gsub("_2", "", x) }))
Bacteroidetes.RA <- data.frame(lapply(Bacteroidetes.RA, function(x) { gsub("_3", "", x) }))
Bacteroidetes.RA[,2] <- as.numeric(as.character(Bacteroidetes.RA[,2]))
Bacteroidetes.RA
#write.csv(Bacteroidetes.RA,"Bacteroidetes.summary.csv")

#Perform ANOVA and TukeyHSD posthoc
Bacteroidetes_aov <- aov(total ~ Group, data = Bacteroidetes.RA)
summary(Bacteroidetes_aov)
Bacteroidetes_tukey <- HSD.test(Bacteroidetes_aov, 'Group')
Bacteroidetes_tukey

#Plot Bacteroidetes relative abundance
p.Bacteroidetes<- ggplot(data=Bacteroidetes.RA, aes(x=Group, y=total, fill=Group))+
  ReplicateMerge +
  geom_boxplot(fill="#919f70")+
  geom_dotplot(binaxis='y',fill="black", stackdir='center', stackratio=1, dotsize=1.25)+
  ylim(0,25)+
  labs(title = "Bacteroidetes", x = "Group", y = "Abundance") +
  theme_bw(base_size = 20) +
  theme(axis.text.x=element_text(size=12,color="black",angle=90), 
        axis.text.y=element_text(size=12,color="black"), 
        axis.title=element_text(size=12,face="bold"),
        text=element_text(size=12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5))
p.Bacteroidetes

#ggsave("Bacteroidetes_sub.pdf", plot = p.Bacteroidetes, width=5, height=6, useDingbats=FALSE)
#ggsave("Bacteroidetes_sub.png", plot = p.Bacteroidetes, width=5, height=6, dpi=600)
```

Gemmatimonadetes RA, Figure 1I
```{r}
#Subset Gemmatimonadetes
all.Gemmatimonadetes <- subset_taxa(all.trans, Phylum %in% c("p__Gemmatimonadetes"))
all.Gemmatimonadetes.ASV.temp <- psmelt(all.Gemmatimonadetes)
all.Gemmatimonadetes.ASV <- subset(all.Gemmatimonadetes.ASV.temp, Abundance > 0)
length(unique(all.Gemmatimonadetes.ASV$OTU))

Gemmatimonadetes.temp <- as.data.frame(paste(all.Gemmatimonadetes.ASV$Group,all.Gemmatimonadetes.ASV$Rep,sep="_"))
Gemmatimonadetes.temp$Abundance <- all.Gemmatimonadetes.ASV$Abundance
Gemmatimonadetes.temp$OTU <- all.Gemmatimonadetes.ASV$OTU
colnames(Gemmatimonadetes.temp)[1] <- "Group"
Gemmatimonadetes.RA <- Gemmatimonadetes.temp%>%
  group_by(Group)%>%
  summarise(total=sum(Abundance))
Gemmatimonadetes.RA
#Remove replicate names from Group
Gemmatimonadetes.RA <- data.frame(lapply(Gemmatimonadetes.RA, function(x) { gsub("_1", "", x) }))
Gemmatimonadetes.RA <- data.frame(lapply(Gemmatimonadetes.RA, function(x) { gsub("_2", "", x) }))
Gemmatimonadetes.RA <- data.frame(lapply(Gemmatimonadetes.RA, function(x) { gsub("_3", "", x) }))
Gemmatimonadetes.RA[,2] <- as.numeric(as.character(Gemmatimonadetes.RA[,2]))
Gemmatimonadetes.RA
#write.csv(Gemmatimonadetes.RA,"Gemmatimonadetes.summary.csv")

#Perform ANOVA and TukeyHSD posthoc
Gemmatimonadetes_aov <- aov(total ~ Group, data = Gemmatimonadetes.RA)
summary(Gemmatimonadetes_aov)
Gemmatimonadetes_tukey <- HSD.test(Gemmatimonadetes_aov, 'Group')
Gemmatimonadetes_tukey

#Plot Gemmatimonadetes relative abundance
p.Gemmatimonadetes<- ggplot(data=Gemmatimonadetes.RA, aes(x=Group, y=total, fill=Group))+
  ReplicateMerge +
  geom_boxplot(fill="#84b59c")+
  geom_dotplot(binaxis='y',fill="black", stackdir='center', stackratio=1, dotsize=1.25)+
  ylim(0,10)+
  labs(title = "Gemmatimonadetes", x = "Group", y = "Abundance") +
  theme_bw(base_size = 20) +
  theme(axis.text.x=element_text(size=12,color="black",angle=90), 
        axis.text.y=element_text(size=12,color="black"), 
        axis.title=element_text(size=12,face="bold"),
        text=element_text(size=12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5))
p.Gemmatimonadetes

#ggsave("Gemmatimonadetes_sub.pdf", plot = p.Gemmatimonadetes, width=5, height=6, useDingbats=FALSE)
#ggsave("Gemmatimonadetes_sub.png", plot = p.Gemmatimonadetes, width=5, height=6, dpi=600)
```

Firmicutes RA, Figure 1F
```{r}
#Subset Firmicutes
all.Firmicutes <- subset_taxa(all.trans, Phylum %in% c("p__Firmicutes"))
all.Firmicutes.ASV.temp <- psmelt(all.Firmicutes)
all.Firmicutes.ASV <- subset(all.Firmicutes.ASV.temp, Abundance > 0)
length(unique(all.Firmicutes.ASV$OTU))

Firmicutes.temp <- as.data.frame(paste(all.Firmicutes.ASV$Group,all.Firmicutes.ASV$Rep,sep="_"))
Firmicutes.temp$Abundance <- all.Firmicutes.ASV$Abundance
Firmicutes.temp$OTU <- all.Firmicutes.ASV$OTU
colnames(Firmicutes.temp)[1] <- "Group"
Firmicutes.RA <- Firmicutes.temp%>%
  group_by(Group)%>%
  summarise(total=sum(Abundance))
Firmicutes.RA
#Remove replicate names from Group
Firmicutes.RA <- data.frame(lapply(Firmicutes.RA, function(x) { gsub("_1", "", x) }))
Firmicutes.RA <- data.frame(lapply(Firmicutes.RA, function(x) { gsub("_2", "", x) }))
Firmicutes.RA <- data.frame(lapply(Firmicutes.RA, function(x) { gsub("_3", "", x) }))
Firmicutes.RA[,2] <- as.numeric(as.character(Firmicutes.RA[,2]))
Firmicutes.RA
#write.csv(Firmicutes.RA,"Firmicutes.summary.csv")

#Perform ANOVA and TukeyHSD posthoc
Firmicutes_aov <- aov(total ~ Group, data = Firmicutes.RA)
summary(Firmicutes_aov)
Firmicutes_tukey <- HSD.test(Firmicutes_aov, 'Group')
Firmicutes_tukey

#Plot Firmicutes relative abundance
p.Firmicutes<- ggplot(data=Firmicutes.RA, aes(x=Group, y=total, fill=Group))+
  ReplicateMerge +
  geom_boxplot(fill="#427e83")+
  geom_dotplot(binaxis='y',fill="black", stackdir='center', stackratio=1, dotsize=1.25)+
  ylim(0,15)+
  labs(title = "Firmicutes", x = "Group", y = "Abundance") +
  theme_bw(base_size = 20) +
  theme(axis.text.x=element_text(size=12,color="black",angle=90), 
        axis.text.y=element_text(size=12,color="black"), 
        axis.title=element_text(size=12,face="bold"),
        text=element_text(size=12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5))
p.Firmicutes

#ggsave("Firmicutes_sub.pdf", plot = p.Firmicutes, width=5, height=6, useDingbats=FALSE)
#ggsave("Firmicutes_sub.png", plot = p.Firmicutes, width=5, height=6, dpi=600)
```

Verrucomicrobia RA
```{r}
#Subset Verrucomicrobia
all.Verrucomicrobia <- subset_taxa(all.trans, Phylum %in% c("p__Verrucomicrobia"))
all.Verrucomicrobia.ASV.temp <- psmelt(all.Verrucomicrobia)
all.Verrucomicrobia.ASV <- subset(all.Verrucomicrobia.ASV.temp, Abundance > 0)
length(unique(all.Verrucomicrobia.ASV$OTU))

Verrucomicrobia.temp <- as.data.frame(paste(all.Verrucomicrobia.ASV$Group,all.Verrucomicrobia.ASV$Rep,sep="_"))
Verrucomicrobia.temp$Abundance <- all.Verrucomicrobia.ASV$Abundance
Verrucomicrobia.temp$OTU <- all.Verrucomicrobia.ASV$OTU
colnames(Verrucomicrobia.temp)[1] <- "Group"
Verrucomicrobia.RA <- Verrucomicrobia.temp%>%
  group_by(Group)%>%
  summarise(total=sum(Abundance))
Verrucomicrobia.RA
#Remove replicate names from Group
Verrucomicrobia.RA <- data.frame(lapply(Verrucomicrobia.RA, function(x) { gsub("_1", "", x) }))
Verrucomicrobia.RA <- data.frame(lapply(Verrucomicrobia.RA, function(x) { gsub("_2", "", x) }))
Verrucomicrobia.RA <- data.frame(lapply(Verrucomicrobia.RA, function(x) { gsub("_3", "", x) }))
Verrucomicrobia.RA[,2] <- as.numeric(as.character(Verrucomicrobia.RA[,2]))
Verrucomicrobia.RA
#write.csv(Verrucomicrobia.RA,"Verrucomicrobia.summary.csv")

#Perform ANOVA and TukeyHSD posthoc
Verrucomicrobia_aov <- aov(total ~ Group, data = Verrucomicrobia.RA)
summary(Verrucomicrobia_aov)
Verrucomicrobia_tukey <- HSD.test(Verrucomicrobia_aov, 'Group')
Verrucomicrobia_tukey

#Plot Verrucomicrobia relative abundance
p.Verrucomicrobia<- ggplot(data=Verrucomicrobia.RA, aes(x=Group, y=total, fill=Group))+
  ReplicateMerge +
  geom_boxplot(fill="#7997a1")+
  geom_dotplot(binaxis='y',fill="black", stackdir='center', stackratio=1, dotsize=1.25)+
  ylim(0,6)+
  labs(title = "Verrucomicrobia", x = "Group", y = "Abundance") +
  theme_bw(base_size = 20) +
  theme(axis.text.x=element_text(size=12,color="black",angle=90), 
        axis.text.y=element_text(size=12,color="black"), 
        axis.title=element_text(size=12,face="bold"),
        text=element_text(size=12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5))
p.Verrucomicrobia
```

Nitrospirae RA
```{r}
#Subset Nitrospirae
all.Nitrospirae <- subset_taxa(all.trans, Phylum %in% c("p__Nitrospirae"))
all.Nitrospirae.ASV.temp <- psmelt(all.Nitrospirae)
all.Nitrospirae.ASV <- subset(all.Nitrospirae.ASV.temp, Abundance > 0)
length(unique(all.Nitrospirae.ASV$OTU))

Nitrospirae.temp <- as.data.frame(paste(all.Nitrospirae.ASV$Group,all.Nitrospirae.ASV$Rep,sep="_"))
Nitrospirae.temp$Abundance <- all.Nitrospirae.ASV$Abundance
Nitrospirae.temp$OTU <- all.Nitrospirae.ASV$OTU
colnames(Nitrospirae.temp)[1] <- "Group"
Nitrospirae.RA <- Nitrospirae.temp%>%
  group_by(Group)%>%
  summarise(total=sum(Abundance))
Nitrospirae.RA
#Remove replicate names from Group
Nitrospirae.RA <- data.frame(lapply(Nitrospirae.RA, function(x) { gsub("_1", "", x) }))
Nitrospirae.RA <- data.frame(lapply(Nitrospirae.RA, function(x) { gsub("_2", "", x) }))
Nitrospirae.RA <- data.frame(lapply(Nitrospirae.RA, function(x) { gsub("_3", "", x) }))
Nitrospirae.RA[,2] <- as.numeric(as.character(Nitrospirae.RA[,2]))
Nitrospirae.RA
#write.csv(Nitrospirae.RA,"Nitrospirae.summary.csv")

#Perform ANOVA and TukeyHSD posthoc
Nitrospirae_aov <- aov(total ~ Group, data = Nitrospirae.RA)
summary(Nitrospirae_aov)
Nitrospirae_tukey <- HSD.test(Nitrospirae_aov, 'Group')
Nitrospirae_tukey

#Plot Nitrospirae relative abundance
p.Nitrospirae<- ggplot(data=Nitrospirae.RA, aes(x=Group, y=total, fill=Group))+
  ReplicateMerge +
  geom_boxplot(fill="#a1c5c8")+
  geom_dotplot(binaxis='y',fill="black", stackdir='center', stackratio=1, dotsize=1.25)+
  ylim(0,5)+
  labs(title = "Nitrospirae", x = "Group", y = "Abundance") +
  theme_bw(base_size = 20) +
  theme(axis.text.x=element_text(size=12,color="black",angle=90), 
        axis.text.y=element_text(size=12,color="black"), 
        axis.title=element_text(size=12,face="bold"),
        text=element_text(size=12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5))
p.Nitrospirae
```

TM7 RA (Saccharibacteria), Figure 1J
```{r}
#Subset TM7
all.TM7 <- subset_taxa(all.trans, Phylum %in% c("p__TM7"))
all.TM7.ASV.temp <- psmelt(all.TM7)
all.TM7.ASV <- subset(all.TM7.ASV.temp, Abundance > 0)
length(unique(all.TM7.ASV$OTU))

TM7.temp <- as.data.frame(paste(all.TM7.ASV$Group,all.TM7.ASV$Rep,sep="_"))
TM7.temp$Abundance <- all.TM7.ASV$Abundance
TM7.temp$OTU <- all.TM7.ASV$OTU
colnames(TM7.temp)[1] <- "Group"
TM7.RA <- TM7.temp%>%
  group_by(Group)%>%
  summarise(total=sum(Abundance))
TM7.RA
#Remove replicate names from Group
TM7.RA <- data.frame(lapply(TM7.RA, function(x) { gsub("_1", "", x) }))
TM7.RA <- data.frame(lapply(TM7.RA, function(x) { gsub("_2", "", x) }))
TM7.RA <- data.frame(lapply(TM7.RA, function(x) { gsub("_3", "", x) }))
TM7.RA[,2] <- as.numeric(as.character(TM7.RA[,2]))
TM7.RA
#write.csv(TM7.RA,"TM7.summary.csv")

#Perform ANOVA and TukeyHSD posthoc
TM7_aov <- aov(total ~ Group, data = TM7.RA)
summary(TM7_aov)
TM7_tukey <- HSD.test(TM7_aov, 'Group')
TM7_tukey

#Plot TM7 relative abundance
p.TM7<- ggplot(data=TM7.RA, aes(x=Group, y=total, fill=Group))+
  ReplicateMerge +
  geom_boxplot(fill="#599ec4")+
  geom_dotplot(binaxis='y',fill="black", stackdir='center', stackratio=1, dotsize=1.25)+
  ylim(0,5)+
  labs(title = "Saccharibacteria", x = "Group", y = "Abundance") +
  theme_bw(base_size = 20) +
  theme(axis.text.x=element_text(size=12,color="black",angle=90), 
        axis.text.y=element_text(size=12,color="black"), 
        axis.title=element_text(size=12,face="bold"),
        text=element_text(size=12)) +
  theme(plot.title = element_text(size = 20, hjust = 0.5))
p.TM7

#ggsave("TM7_sub.pdf", plot = p.TM7, width=5, height=6, useDingbats=FALSE)
#ggsave("TM7_sub.png", plot = p.TM7, width=5, height=6, dpi=600)
```

Here is code that can make a single merged plot with all phyla subsets, using cowplot. 
```{r eval=FALSE, include=TRUE}
(p.Actino|p.Proteo |p.Acido |p.Chloroflexi |p.Bacteroidetes )/(p.Gemmatimonadetes |p.Firmicutes |p.Verrucomicrobia |p.Nitrospirae |p.TM7 )
# ggsave does not work with cowplot. Export with width=16, height= 9.
```

## Metabolomics profiling, TP8 only, Figure 2
import data and perform PCA
```{r}
pca_tp8 <- read.csv("pca_tp8_metabolites.csv", header = FALSE) #TP8 subset of merged dataset with transposed rows
log.tp8 <- log(pca_tp8[,3:170])
tp8.group <- pca_tp8[1:18,2]

# apply PCA - scale. = TRUE is highly advisable, but default is FALSE. 
tp8.pca <- prcomp(log.tp8,
                  center = TRUE,
                  scale. = TRUE) 
# plot method
plot(tp8.pca, type = "l")
```

Generate a PCA biplot in base R, Figure 2B
```{r}
autoplot(tp8.pca) #Use this default plot to determine the X and Y axis %
p<-tp8.pca
pch.group <- c(rep(24, times=3), rep(21, times=3),
               rep(24, times=3), rep(21, times=3),
               rep(24, times=3), rep(21, times=3))

col.group <- c(rep("#919f70", times=3), rep("#919f70", times=3),
               rep("#7997a1", times=3), rep("#7997a1", times=3),
               rep("#ecc363", times=3), rep("#ecc363", times=3))

#Create the plot used for manuscript figure
plot(p$x[,1], p$x[,2], xlab=paste("PCA 1 (","59.5", "%)",sep = ""), #check this number with autoplot
     ylab=paste("PCA 2 (","17.1", "%)",sep = ""), #check this number with autoplot
     pch=pch.group, col="black", bg=col.group, cex=5, las=1, asp=1,
     panel.first= {abline(v=15, lty=1, col="light grey")
       abline(v=10, lty=1, col="light grey")
       abline(v=5, lty=1, col="light grey")
       abline(v=0, lty=5, col="dark grey")
       abline(v=-5, lty=1, col="light grey")
       abline(v=-10, lty=1, col="light grey")
       abline(v=-15, lty=1, col="light grey")
       abline(h=10, lty=1, col="light grey")
       abline(h=5, lty=1, col="light grey")
       abline(h=0, lty=5, col="dark grey")
       abline(h=-5, lty=1, col="light grey")
       abline(h=-10, lty=1, col="light grey")
     })
#Run the following code to place legend on top of plot
#legend("topleft", legend=c("TP8-W","TP8-D",
#                           "TP8-W", "TP8-D", 
#                           "TP8-W","TP8-D"), col="black", 
#       pt.bg=c("#919f70", "#919f70",
#               "#7997a1", "#7997a1",
#               "#ecc363", "#ecc363"), 
#       pch=c(24, 21, 24, 21, 24, 21), pt.cex=2)
```

## Metabolomics profiling, TP8 and 24h, Figure 3
import data and perform PCA
```{r}
pca_all <- read.csv("pca_all_metabolites.csv", header = FALSE) #merged full dataset with transposed rows
log.all <- log(pca_all[,3:170])
all.group <- pca_all[1:36,2]

# apply PCA - scale. = TRUE is highly advisable, but default is FALSE. 
all.pca <- prcomp(log.all,
                 center = TRUE,
                 scale. = TRUE) 
# plot method
plot(all.pca, type = "l")
```

Generate a PCA biplot in base R, Figure 3D
```{r}
autoplot(all.pca) #Use this default plot to determine the X and Y axis %

p<-all.pca
pch.group <- c(rep(24, times=3), rep(21, times=3),rep(24, times=3),rep(21, times=3),
               rep(24, times=3), rep(21, times=3),rep(24, times=3),rep(21, times=3),
               rep(24, times=3), rep(21, times=3),rep(24, times=3),rep(21, times=3))

col.group <- c(rep("#919f70", times=3), rep("#919f70", times=3),rep("#686a47", times=3),rep("#686a47", times=3),
               rep("#7997a1", times=3), rep("#7997a1", times=3),rep("#4d5b6a", times=3),rep("#4d5b6a", times=3),
               rep("#ecc363", times=3), rep("#ecc363", times=3),rep("#cb9e59", times=3),rep("#cb9e59", times=3))

#Create the plot used for manuscript figure
plot(p$x[,1], p$x[,2], xlab=paste("PCA 1 (","55.52", "%)",sep = ""), #match this number with autoplot results above
                ylab=paste("PCA 2 (","14.74", "%)",sep = ""), #match this number with autoplot results above
                pch=pch.group, col="black", bg=col.group, cex=5, las=1, asp=1,
                panel.first= {abline(v=15, lty=1, col="light grey")
                              abline(v=10, lty=1, col="light grey")
                              abline(v=5, lty=1, col="light grey")
                              abline(v=0, lty=5, col="dark grey")
                              abline(v=-5, lty=1, col="light grey")
                              abline(v=-10, lty=1, col="light grey")
                              abline(v=-15, lty=1, col="light grey")
                              abline(h=10, lty=1, col="light grey")
                              abline(h=5, lty=1, col="light grey")
                              abline(h=0, lty=5, col="dark grey")
                              abline(h=-5, lty=1, col="light grey")
                              abline(h=-10, lty=1, col="light grey")
                              })
#Run the following code to place legend on top of plot
#legend("topleft", legend=c("TP8-W","24h-W", "TP8-D",  "24h-DW",
#                           "TP8-W","24h-W", "TP8-D",  "24h-DW",
#                           "TP8-W","24h-W", "TP8-D",  "24h-DW"), col="black", 
#       pt.bg=c("#919f70", "#686a47", "#919f70", "#686a47",
#               "#7997a1", "#4d5b6a", "#7997a1", "#4d5b6a",
#               "#ecc363", "#cb9e59", "#ecc363", "#cb9e59"), 
#       pch=c(24,24, 21,21, 24, 24, 21, 21, 24, 24, 21, 21), pt.cex=2)
```
